### PR TITLE
Allow blake2 & parity-codec in trie

### PR DESCRIPTION
**Pull request by jacogr**
Requires/contains https://github.com/polkadot-js/common/pull/256

### CODE REVIEW

File: packages/trie-codec/src/NodeHeader.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    throw new Error('Unreachable');
+  }
+
+  get encodedLength (): number {
+    const nodeType = this.nodeType;
+
+    if (nodeType === NODE_TYPE_NULL || nodeType === NODE_TYPE_BRANCH) {
+      return 1;
+    } else if (nodeType === NODE_TYPE_EXT) {
+      const nibbleCount = (this.value.raw as ExtensionHeader).toNumber();
```
Comment (amaurymartiny) [inline packages/trie-codec/src/NodeHeader.ts]: `this.value` is an ExtensionHeader already, I think you don't need raw here

File: packages/chainspec/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                              Apache License
```
Comment (jacogr) [inline packages/chainspec/LICENSE]: Package created so there is a common place where the chainspecs can be held. Currently these are actually included in the client. Putting it here (and they don't update frequently) has the added benefit that we can actually use actual data in tests, for instance to calculate the trie-roots and then compare against known.

File: packages/trie-hash/test/data.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
+// Copyright 2017-2018 @polkadot/trie-hash authors & contributors
```
Comment (jacogr) [inline packages/trie-hash/test/data.ts]: These are basically the Rust tests - as a packaged file so we can use it both in trie-hash and trie-db to check the results

File: packages/util/src/compact/addLength.spec.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2017-2018 @polkadot/util authors & contributors
```
Comment (jacogr) [inline packages/util/src/compact/addLength.spec.ts]: types/codec/Compact.statics follow - api/types to be updated to use these

File: packages/util/src/u8a/toU8a.spec.js [hunk 5]
```diff
@@ -33,6 +33,14 @@ describe('u8aToU8a', () => {
     );
   });
 
+  it('creates from Array', () => {
```
Comment (jacogr) [inline packages/util/src/u8a/toU8a.spec.js]: It was uncovered... weak moment for my OCD.

**Pull request merged**